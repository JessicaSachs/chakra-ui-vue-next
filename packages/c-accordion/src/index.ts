import { h, defineComponent, PropType } from 'vue'
import { chakra, DOMElements } from '@chakra-ui/vue-system'

const CAccordion = defineComponent({
  props: {
    as: {
      type: [String] as PropType<DOMElements>,
      default: 'div',
    },
  },
  setup(props, { slots, attrs }) {
    return () => h(chakra(props.as), { ...attrs }, slots)
  },
})

export default CAccordion
