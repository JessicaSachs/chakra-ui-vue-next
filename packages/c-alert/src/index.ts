import { h, defineComponent, Component } from 'vue'

const CAlert = defineComponent({
  props: {
    as: {
      type: [String, Object],
      default: 'div',
    },
  },
  render() {
    return h(
      this.as,
      {
        ...this.$attrs,
        role: 'alert',
      },
      this.$slots.default && this.$slots.default()
    )
  },
})

export default CAlert
