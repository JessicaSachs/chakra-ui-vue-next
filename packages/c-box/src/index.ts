import { h, defineComponent, PropType } from 'vue'

const CBox = defineComponent({
  props: {
    as: {
      type: String as PropType<string>,
      default: 'div',
    },
  },
  render() {
    return h(this?.as, { ...this.$props, ...this.$attrs }, this.$slots)
  },
})

export default CBox
