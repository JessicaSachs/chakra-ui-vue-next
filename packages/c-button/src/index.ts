import { h, defineComponent, PropType } from 'vue'

const CButton = defineComponent({
  name: 'CButton',
  props: {
    as: {
      type: String as PropType<string>,
      default: 'button',
    },
  },
  render() {
    return h(
      this?.as,
      { ...this.$props, ...this.$attrs },
      this.$slots.default && this.$slots.default()
    )
  },
})

export default CButton
