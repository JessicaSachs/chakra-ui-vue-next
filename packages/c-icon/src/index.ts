export * from './icon'
export { default as internalIcons } from './icon.internals'
export * from './icon.internals'
