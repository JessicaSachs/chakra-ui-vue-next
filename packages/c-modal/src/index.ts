export { default as CModal } from './c-modal'
