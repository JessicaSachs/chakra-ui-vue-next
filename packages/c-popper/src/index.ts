export { default as usePopper } from './vue-popper'
