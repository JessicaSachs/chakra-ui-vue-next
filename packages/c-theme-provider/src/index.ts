import {
  h,
  defineComponent,
  Fragment,
  PropType,
  provide,
  inject,
  computed,
} from 'vue'
import { ThemeProviderProps } from '@chakra-ui/vue-next'

const CThemeProvider = defineComponent({
  name: 'CThemeProvider',
  props: {
    value: {
      type: [Object] as PropType<ThemeProviderProps>,
      default: () => undefined,
    },
  },
  setup(props, { slots }) {
    const pluginTheme = inject('$chakraTheme')
    const applicationTheme = computed(() => props.value || pluginTheme)
    provide('$chakraTheme', applicationTheme.value)
    return () => h(Fragment, slots.default?.({ $chakraTheme: props.value }))
  },
})

export default CThemeProvider
