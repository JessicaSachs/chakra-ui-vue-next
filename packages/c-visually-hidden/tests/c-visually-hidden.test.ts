import { render } from '../../test-utils/src'
import { CVisuallyHidden } from '../src'

it('should render properly', () => {
  const { asFragment } = render(CVisuallyHidden)
  expect(asFragment()).toMatchSnapshot()
})