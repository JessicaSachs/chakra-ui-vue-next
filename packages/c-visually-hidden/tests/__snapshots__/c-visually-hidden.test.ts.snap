// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <span
    class="css-f8n5zr"
  />
</DocumentFragment>
`;
